cubed/array_api/creation_functions.py

```diff
@@ -24,7 +25,12 @@ def arange(
         start, stop = 0, start
     num = int(max(math.ceil((stop - start) / step), 0))
     if dtype is None:
+        # TODO: Use inspect API
         dtype = nxp.arange(start, stop, step * num if num else step).dtype
+        for k, dtype_ in default_dtypes(device=device).items():
+            if nxp.isdtype(dtype, k):
+                dtype = dtype_
```

Comment:
It's not clear to me what this is doing or why it is needed. If nxp is the jax namespace doesn't the call to arange already return the correct dtype (int32) - or does jax metal just return int64 or fail?

It might help to factor out a function to do this (given it is duplicated below too) with a name describing what it does, and perhaps a comment too.

Reply:
> or does jax metal just return int64 or fail?
Yes, it looks like jax metal returns 64 bit precisions and fails; this ensures the correct precision. 

I've factored this out to a function, good shout.